tasks/main.yml

```diff
@@ -16,8 +16,31 @@
     - node_exporter_configure
     - node_exporter_run
 
+- name: Check if node_exporter is installed
+  stat:
+    path: "{{ node_exporter_destination }}"
+  register: __node_exporter_is_installed
+  tags:
+    - node_exporter_install
+
+- name: See stat (REMOVE THIS LATER)
+  debug:
+    msg: "{{ __node_exporter_is_installed }}"
+
+- name: Gather currently installed node_exporter version (if any)
+  command: "{{ node_exporter_destination }} --version"
+  args:
+    warn: false
+  changed_when: false
+  register: __node_exporter_current_version_output
```

Comment:
I wonder what will happen if we have more than one target host.

Reply:
@paulfantom you mean, a mix of "has it installed" and "doesn't have it installed"? I was _assuming_ the tests running against various distros would account for that, it just runs on every hosts (no `run_once` or so).

Reply:
I mean, `command` output for one host has different structure than for many hosts. 

Also in ideal world we would need to copy binary only to hosts which doesn't have version we need. But for now I would suggest to go with a naive implementation and if at least one host has wrong version then propagate new one to all hosts.

Reply:
You are right, it should work :+1: